internal/core/command/types_test.go

```diff
@@ -67,41 +68,50 @@ func TestExecute(t *testing.T) {
 
 	request, _ := http.NewRequest(http.MethodGet, ts.URL, nil)
 	sc := newServiceCommand(contract.Device{AdminState: contract.Unlocked}, &http.Client{}, request, logger.NewMockClient())
+	deviceServiceResponse, err := sc.Execute()
 
-	body, responseCode, err := sc.Execute()
 	if err != nil {
 		t.Errorf("No error should be present for happy path")
 	}
 
-	if body != expectedResponseBody {
-		t.Errorf("The response body was not properly propegated to the caller")
+	// Extract the headers from the device service's response.
+	headers := map[string]string{"Content-Type": ""}
+	m := make(map[string][]string)
+	m = deviceServiceResponse.Header
+	for name, values := range m {
+		for _, value := range values {
+			headers[name] = value
+		}
 	}
```

Comment:
As a follow up to this issue and not to be handled now, I think we need to clean this up. Using names like `m` makes this hard to read and --as we seen in the iterations on this PR-- causes bugs that are only evident during manual testing.